sdk/core/core-util/src/createAbortablePromise.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortError, AbortSignalLike } from "@azure/abort-controller";
+
+/**
+ * Creates an abortable promise.
+ * @param buildPromise - A function that takes the resolve and reject functions as parameters.
+ * @param options - The options for the abortable promise.
+ * @returns A promise that can be aborted.
+ */
+export function createAbortablePromise<T>(
+  buildPromise: (
+    resolve: (value: T | PromiseLike<T>) => void,
+    reject: (reason?: any) => void
+  ) => void,
+  options?: {
+    /** A function to be called if the promise was aborted */
+    cleanupBeforeAbort?: () => void;
+    /** An abort signal */
+    abortSignal?: AbortSignalLike;
+    /** An abort error message */
+    abortErrorMsg?: string;
```

Comment:
nit: would you mind making this a type to help the readability of the signature?

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-js/pull/24731/commits/8f152da300ad3d8bf963184baf35d517ff2ebf8a.